### PR TITLE
[CALCITE-4342] More aggregate functions should be marked as splittable and ignore distinct optionality

### PR DESCRIPTION
**Pull request by liyafan82**
Please see the discussions in https://issues.apache.org/jira/browse/CALCITE-4342

### CODE REVIEW

File: core/src/main/java/org/apache/calcite/sql/SqlAggFunction.java [hunk 1]
```diff
@@ -204,4 +204,11 @@ public boolean allowsFilter() {
   public boolean allowsNullTreatment() {
     return false;
   }
+
+  /**
+   * Get rollup aggregation function.
+   */
+  public SqlAggFunction getRollup() {
```
Comment (chunweilei) [inline core/src/main/java/org/apache/calcite/sql/SqlAggFunction.java:211]: Get -> Gets

Comment (liyafan82) [reply core/src/main/java/org/apache/calcite/sql/SqlAggFunction.java:211]: Fixed. Thank you.

**Review by chunweilei — COMMENTED** [review]
LGTM

**Review by chunweilei — APPROVED** [review]

**Comment by liyafan82** [comment]
If there are no objections, I want to merge this PR in a few days.

**Comment by liyafan82** [comment]
Merging. Thanks a lot for your comments. @chunweilei

**Pull request merged**